### PR TITLE
8-healing-resource-pages

### LINKED ISSUE #8: Healing Resource Pages
## Todo: ##
- On the healing resources page, create 2 buttons that lead to different pages -> one for general healing resources, and the other should lead to the HFH guide. 
- The `Healing Catalogue` screen should look very similar to Cody's sprint task, basically you'll be rendering a `List` of resources based on theme, but for now, just have dummy buttons on the side, and render the entire list of resources 
- The other screen should lead to the Hope for Healing guide, which for now, should just display a screen that says "hello". More of this on the next sprint

image

## Useful Links: ##
#7

### PR DESCRIPTION
**Pull request by arfamomin**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Main healing resources page now has two functional buttons
- Healing Catalogue is now populated with resources from supabase querying (Resources that fit multiple categories are also populated as such)
- Left nav buttons control navigation of themes on scroll 

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
IMG_0881
IMG_0882

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- Review all at once, most work done in "HealingCatalogue" folder 
- Also review if I did the supabase querying and types stuff correctly (this is my first time doing it so def correct me if I did it wrong)

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- I did not end up creating a new component for a "resource card" yet because not sure the complete props to give it yet
- The themed nav button styling isnt exactly like the mid-fi so I think that can be cleaned up later (maybe developed in the HFH sprint?)

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

https://reactnative.dev/docs/scrollview (Possibly check if there is an easier way than I did it, which was storing the y position of the header as it page gets rendered)

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
 N/A for now

CC: @philipye314

### CODE REVIEW

File: src/screens/HealingResources/HealingCatalogue/index.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            <Text>{theme}</Text>
+          </TouchableOpacity>
+        ))}
+      </View>
+      <ScrollView ref={scrollViewRef}>
+        <View style={styles.resourcesContainer}>
+          {themes.map((theme, index) => (
+            <>
+              <View style={styles.themeHeader} onLayout={onLayoutHeader(index)}>
+                <Text style={styles.themeHeaderText}>{theme}</Text>
```
Comment (philipye314) [inline src/screens/HealingResources/HealingCatalogue/index.tsx]: add a key here to stop typescript from yelling lol

Comment (philipye314) [reply src/screens/HealingResources/HealingCatalogue/index.tsx]: done

**Review by philipye314 — DISMISSED** [review]
Looks good! I fixed some really minor things, great work

**Review by philipye314 — APPROVED** [review]
APPROVED BRUH
